Source/AwsCommonRuntimeKit/auth/imds/CRTIMDSInstanceInfo.swift

```diff
@@ -5,20 +5,20 @@ import AwsCAuth
 import Foundation
 
 public struct CRTIMDSInstanceInfo {
-    public var marketPlaceProductCodes: [String]
-    public var availabilityZone: String?
-    public var privateIp: String?
-    public var version: String?
-    public var instanceId: String?
-    public var billingProducts: [String]
-    public var instanceType: String?
-    public var accountId: String?
-    public var imageId: String?
-    public var pendingTime: Date
-    public var architecture: String?
-    public var kernelId: String?
-    public var ramDiskId: String?
-    public var region: String?
+    public let marketPlaceProductCodes: [String]
+    public let availabilityZone: String
+    public let privateIp: String
+    public let version: String
+    public let instanceId: String
+    public let billingProducts: [String]
+    public let instanceType: String
+    public let accountId: String
+    public let imageId: String
+    public let pendingTime: Date
+    public let architecture: String
+    public let kernelId: String
+    public let ramDiskId: String
+    public let region: String
```

Comment:
They don’t have to be always present, the only difference is whether we want to return nil or “” for empty value. Looking at the previous code, we were replacing nil with “”. The C code fetches a json from the instance, and the instance might have “” as the json value. The C code treats “” as a valid byte cursor. I thought it is a better API design to keep “” as empty instead of Nil.